kong/router/atc.lua

```diff
@@ -158,35 +158,31 @@ local function add_atc_matcher(inst, route, route_id,
 end
 
 
-local function is_http_headers_field(field)
-  return field:sub(1, 13) == "http.headers."
-end
-
+local function categorize_fields(fields)
 
-local function has_header_matching_field(fields)
-  for _, field in ipairs(fields) do
-    if is_http_headers_field(field) then
-      return true
-    end
+  if not is_http then
+    return fields, nil, nil
   end
 
-  return false
-end
+  local basic = {}
+  local headers = {}
+  local queries = {}
 
+  for _, field in ipairs(fields) do
+    local prefix = field:sub(1, 13)
 
-local function is_http_queries_field(field)
-  return field:sub(1, 13) == "http.queries."
-end
+    if prefix == "http.headers." then
+      headers[field:sub(14)] = field
 
+    elseif prefix == "http.queries." then
+      queries[field:sub(14)] = field
```

Comment:
Yes, `13` and `14` are a little weird, but defining constants perhaps is another noise. 

Since the prefix is short and easy to understand, I tend to keep it.

If you have a better idea to improve it please create a PR, thanks.